mnist/autoencoder.jl

```diff
@@ -0,0 +1,46 @@
+using Flux
+using Flux: onehotbatch, argmax, mse, throttle, accuracy
+using Base.Iterators: partition
+using MNIST
+
+# Encode MNIST images as compressed vectors that can later be decoded back into
+# images.
+
+x, _ = traindata()
+x ./= 255
+
+# Partition into batches of size 1000
+data = [(x[:,i],) for i in partition(1:60_000, 1000)]
+
+N = 32 # Size of the encoding
+
+m = Chain(
+  Dense(28^2, N, relu), # Encoder
+  Dense(N, 28^2, relu)) # Decoder
```

Comment:
The reason why I separate the encoder and decoder as different variables is that I want user store them (into JLD/serialized file), sometimes the useful part is encoder, sometime is the decoder.

Reply:
Makes sense. You can still do this as `m[1], m[2]`, though it might be clearer to use the names -- feel free to make that patch!